### PR TITLE
Refactoring TableFooter

### PR DESCRIPTION
**Pull request by wdh2100**
Refactor from class component to functional component

ref : #1346

Sorry. When I split it in PR, it's lot..

codeSandBox : http://codesandbox.io/s/github/wdh2100/mui-datatables/tree/functional/tableFooter

### CODE REVIEW

File: test/MUIDataTablePagination.test.js [hunk 1]
```diff
@@ -1,9 +1,7 @@
 import React from 'react';
-import { spy, stub } from 'sinon';
```
Comment (wdh2100) [inline test/MUIDataTablePagination.test.js:2]: unnecessary codes

**Comment by patorjk** [comment]
@wdh2100 Can you verify that you ran "npm run test"? I had some issues but realized it they were already in the v3.1.0 branch (so not related to you). It's kind of late, so I'm thinking of saving the rest of these for next week's release. However, looks like I can't test from codesandbox so figured I'd ask before doing the merge.

**Comment by wdh2100** [comment]
What is the problem?
When I run '**npm run test**' it works fine

asdasd

**Comment by wdh2100** [comment]
but mocha.opts <--- deprecated

https://stackoverflow.com/questions/60283197/mocha-opts-deprecated-how-to-migrate-to-package-json

**Comment by wdh2100** [comment]
Rebase v3.2.0

**Comment by patorjk** [comment]
Thank you for adding the tests. I have a footer feature I want to add for version 3.2.0. I'm going to merge this branch into the jumpToPage branch and work on it there.

**Comment by wdh2100** [comment]
@patorjk 

thanks!

**Pull request merged**